source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,723 @@
+  you can write file contents
+- ``UploadFromStreamAsync()`` or ``UploadFromStream()``: Uploads the contents of an existing
+  stream to a GridFS file
+
+The following sections describe how to use these methods.
+
+.. _csharp-gridfs-open-upload-stream:
+
+Write to an Upload Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``OpenUploadStreamAsync()`` or ``OpenUploadStream()`` method to create an upload
+stream for a given file name. These methods accept the following parameters:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``filename``
+     - The name of the file to upload.
+        
+       **Data type**: {+string-data-type+}
+   * - ``options``
+     - *Optional.* An instance of the ``GridFSUploadOptions`` class that specifies the
+       configuration for the upload stream. The default value is ``null``.
+        
+       **Data type**: `GridFSUploadOptions <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.GridFS.GridFSUploadOptions.html>`__
+   * - ``cancellationToken``
+     - *Optional.* A token that you can use to cancel the operation.
+        
+       **Data type**: `CancellationToken <https://learn.microsoft.com/dotnet/api/system.threading.cancellationtoken>`__
+
+This code example demonstrates how to open an upload stream by performing the
+following steps:
+
+- Calls the ``OpenUploadStream()`` method to open a writable GridFS stream for a file
+  named ``"my_file"``
+- Calls the ``Write()`` method to write data to ``my_file``
+- Calls the ``Close()`` method to close the stream that points to ``my_file`` 
+
+Select the :guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: open-upload-stream-async
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-open-upload-stream-async
+         :end-before: // end-open-upload-stream-async
+
+   .. tab:: Synchronous
+      :tabid: open-upload-stream-sync
+
+      .. literalinclude:: /includes/code-examples/GridFS.cs
+         :language: csharp
+         :dedent:
+         :start-after: // start-open-upload-stream
+         :end-before: // end-open-upload-stream
+
+To customize the upload stream configuration, pass an instance of the
+``GridFSUploadOptions`` class to the ``OpenUploadStreamAsync()`` or ``OpenUploadStream()``
+method. The ``GridFSUploadOptions`` class contains the following properties:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``BatchSize``
+     - The number of chunks to upload in each batch. The default value is ``16777216``
+       divided by the value of the ``ChunkSizeBytes`` property. 
+        
+       **Data type**: ``int?`` 
+       
+   * - ``ChunkSizeBytes``
+     - The maximum size of each chunk.
```

Comment:
maximum?

All chunks except the last one will be exactly this size. Only the last one will be smaller.

Reply:
So it's technically true that this is the maximum size, but that's not the most helpful information to a user!

Reply:
I would not call this the maximum size. It is THE size.

Talking about the maximum implies to me that the chunk size is variable, and it's not. All the chunks (except the last one) are the SAME size.